easybuild/easyblocks/f/fsl.py

```diff
@@ -63,15 +63,43 @@ def configure_step(self):
         fslmachtype = out.strip()
         self.log.debug("FSL machine type: %s" % fslmachtype)
 
-        # prepare config
-        # either using matching config, or copy closest match
+        # Check if a specific machine type directory is patched
+        systype_regex = re.compile(".*(apple|gnu|i686|linux|spark).*", re.M)
+        diff_regex = re.compile("^diff.*config\/", re.M)
+        
+        patched_cfgs = []
+        best_cfg = None
+
+        for patch in self.patches:
+            try:
+                for line in open(patch['path'], 'r'):
+                    if all(regex.match(line) for regex in [systype_regex, diff_regex]):
```

Comment:
The reason they are separate is because I use `systype_regex` twice and only once in conjunction with `diff_regex`. Hence I preferred to split them like that instead of having `systype_regex` being a subset of `diff_regex` (i.e. avoid copy/paste).